### PR TITLE
Updates from near-api-js

### PR DESCRIPTION
**Pull request by volovyk-s**

### CODE REVIEW

**Review by ailisp — COMMENTED** [review]
Is there specific reason to do these renames? I'm curious about why rename here instead of export this and  import and change usage on near-api-js side, since camelCase is more common in js

**Review by frol — APPROVED** [review]
I have the same concern about the snake case in JS code (while I prefer it, I still follow the style that is common in the target language, and camelCase is what used in JS).

@vgrichina Also, I wonder if we can remove `borsh-js` build from the repo? I usually find it to be a source of merge conflicts, bloating the sources, and inconsistencies if contributors forget to build or include the generated files in their PR.

File: borsh-ts/test/serialize.test.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const borsh = require('../lib/index.js');
+const borsh = require('../../borsh-js/index.js');
```
Comment (frol) [inline borsh-ts/test/serialize.test.js:1]: This looks like a hack. Can we use `ts`? Can we write the tests in TS?

**Comment by frol** [comment]
@volovyk-s I want to point out that this repo also contains the Rust implementation of Borsh, and thus `lib` folder in the root is a bit misleading.

@ailisp @evgenykuzyakov @nearmax Are there any objections to renaming this repo to `borsh-rs`, extracting the JS/TS implementation into `borsh-js` repository, and the landing page to `borsh` repository?

**Comment by volovyk-s** [comment]
> Is there specific reason to do these renames? I'm curious about why rename here instead of export this and import and change usage on near-api-js side, since camelCase is more common in js

@ailisp @frol Yep, we are using camel case everywhere, so I have changed it back. I'm not sure why these changes were added in `near-api-js`

**Comment by volovyk-s** [comment]
> @volovyk-s I want to point out that this repo also contains the Rust implementation of Borsh, and thus `lib` folder in the root is a bit misleading.
> 
> @ailisp @evgenykuzyakov @nearmax Are there any objections to renaming this repo to `borsh-rs`, extracting the JS/TS implementation into `borsh-js` repository, and the landing page to `borsh` repository?

It was renamed because `borsh-js` is also confusing. It's not a separate JS implementation, it's just a build artifact of TS subproject. BTW, @frol , JS artifacts can not be deleted, because some developers can use this repo instead of NPM package [(like here).](https://glebbahmutov.com/blog/use-github-instead-of-npm/)  

It would be nice to move TS/JS version to a separate repo.
Pros:
- no confusions with `lib` or `borsh-js` folders (thnx @frol )
- it's confusing for the Rust developers to see ~10 unrelated files in the root directory
- separated CI checks
- separate `README.md` in the root folder (easier onboarding of new users)
Cons:
- partially destroyed Git history

**Comment by frol** [comment]
> Cons: partially destroyed Git history

Git history can be preserved if we just clone this repo and clean up unrelated parts as a regular commit. Yes, we will have borsh-rs in the history in two places, but overall it should be fine.

**Comment by ailisp** [comment]
I also vote for separate borsh-rs

**Review by ailisp — APPROVED** [review]

**Comment by vgrichina** [comment]
> > Cons: partially destroyed Git history
> 
> Git history can be preserved if we just clone this repo and clean up unrelated parts as a regular commit. Yes, we will have borsh-rs in the history in two places, but overall it should be fine.

can be done even better with `git-filter-branch` https://manishearth.github.io/blog/2017/03/05/understanding-git-filter-branch/ (will result in smaller repo, but history essentially will be rewritten)

File: .eslintrc.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    - error
+    - unix
+  quotes:
+    - error
+    - single
+  semi:
+    - error
+    - always
+  no-console: off
+globals:
```
Comment (vgrichina) [inline .eslintrc.yml:22]: @volovyk-s let's keep only actually necessary globals for borsh

**Comment by frol** [comment]
@volovyk-s May I ask you to extract the relevant history into https://github.com/near/borsh-js and https://github.com/near/borsh-rs? (Just push `master` branches there and I will switch the default from `main` to `master` branch on Github)

**Comment by volovyk-s** [comment]
> @volovyk-s May I ask you to extract the relevant history into https://github.com/near/borsh-js and https://github.com/near/borsh-rs? (Just push `master` branches there and I will switch the default from `main` to `master` branch on Github)

Ok, I will do that after this merge.

**Pull request merged**